app/models/pawn.rb

```diff
@@ -1,6 +1,29 @@
 class Pawn < Piece
-  def move_valid?
-    # add logic here
+  def move_valid?(x, y)
+    return false if space_occupied?(x, y) == true
+    return false if legal_move?(x, y) == false
```

Comment:
It looks like the only thing that `legal_move?` is doing is calling `one_step_legal?`, so most likely it should just be removed and this line here can directly call `one_step_legal?`.